Automattic/wp-calypso
# Refactor twitter preview component and update share preview

Related to #75426

## Proposed Changes

* Break `TwitterPreview` and `Tweet` class components into smaller TS functional components
* Update calypso share preview for Twitter to use the updated preview component
* Update linkedin preview to render the image only if present

### Prerequisites
- Make sure you have a Twitter account.
- Make sure you have a self-hosted site connected to Jetpack, with a subscription that includes Social.
- From the _Social_ or _My Jetpack_ page of your site WPadmin, activate Social.
- Then, connect your site to Twitter (check [these instructions](https://jetpack.com/support/jetpack-social/connecting-to-social-networks/) if you need help). You may need to create a test page.
- Spin up Calypso, by running this branch locally or using a live link below.

### Testing

- Create a new post from your site `wp-admin`
- Add a featured image (do not add a custom media)
- Publish it
- Visit `/posts/:site` in Calypso
- You should see the newly created post under the _Published_ tab
- In the post menu, click _Share_, then press the _Preview_ button
- Select the Twitter preview
- You should see a preview of how the post looks when shared

| BEFORE | AFTER |
|--------|--------|
| <img width="1704" alt="Screenshot 2023-04-20 at 3 35 33 PM" src="https://user-images.githubusercontent.com/18226415/233334189-eef4a02b-3849-4cab-99cc-91ab7199d29c.png"> |<img width="1703" alt="Screenshot 2023-04-20 at 3 34 40 PM" src="https://user-images.githubusercontent.com/18226415/233334289-32aec716-e581-4989-8b9c-bb6315b22e1a.png"> |


## Pre-merge Checklist
